include/deal.II/base/function_parser.h

```diff
@@ -328,10 +328,12 @@ class FunctionParser : public AutoDerivativeFunction<dim>
 
   /**
    * The muParser objects for each thread (and one for each component). We are
-   * storing unique_ptr so that we don't need to include the definition of
+   * storing shared_ptr so that we don't need to include the definition of
    * mu::Parser in this header.
+   *
+   * std::unique_ptr does not compile with gcc-5.4
```

Comment:
users probably don't need this information as a part of documentation, so i would remove it.